complete/src/main/java/com/example/restservice/controller/advices/SecurityAdvice.java

```diff
@@ -0,0 +1,66 @@
+package com.example.restservice.controller.advices;
+
+import com.auth0.jwk.JwkException;
+import com.auth0.jwk.JwkProvider;
+import com.auth0.jwk.UrlJwkProvider;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.SignatureVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.interfaces.RSAPublicKey;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.MethodParameter;
+import org.springframework.http.HttpInputMessage;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.servlet.mvc.method.annotation.RequestBodyAdviceAdapter;
+
+@ControllerAdvice
+public class SecurityAdvice extends RequestBodyAdviceAdapter {
+
+  private final JwkProvider provider;
+  private final LoggedInUserInfo loggedInUserInfo;
+
+  @Autowired
+  SecurityAdvice(LoggedInUserInfo loggedInUserInfo) throws MalformedURLException {
+    this.loggedInUserInfo = loggedInUserInfo;
+    String keyUrl = "https://cognito-idp.ap-southeast-1.amazonaws.com/ap-southeast-1_iQdl5AVrA/.well-known/jwks.json";
```

Comment:
Hmm might need a better url later?

Reply:
this is fully internal, so would be fine.

Reply:
Maybe keep this in a constants file then?